test/unit/v_all/test_api.py

```diff
@@ -22,6 +22,84 @@
 from ..test_base import TestBase
 
 
+class TestServices(TestBase):
+    def setUp(self):
+        self.account_info = InMemoryAccountInfo()
+        self.cache = InMemoryCache()
+
+    @pytest.mark.apiver(from_ver=2)
+    def test_api_initialization(self):
+        class DummyA:
+            def __init__(self, *args, **kwargs):
+                pass
+
+        class DummyB:
+            def __init__(self, *args, **kwargs):
+                pass
+
+        for kwargs, _raw_api_class in [
```

Comment:
any reason not do this via `parametrize`? purest doesn't support `with self.subTest` so asserts in loops in tests decrease readability, you have to guess which iteration failed from expected values. `parametrize` fixes that